src/SmsLength.php

```diff
@@ -162,12 +162,17 @@ private function inspect(string $messageContent): void
         // Any character outside the 7-bit alphabet switches the entire encoding to UCS-2
         $this->encoding = '7-bit';
         $this->size = 0;
+
         $mbLength = mb_strlen($messageContent, 'UTF-8');
         for ($i = 0; $i < $mbLength; $i++) {
             $char = mb_substr($messageContent, $i, 1, 'UTF-8');
             if (in_array($char, self::GSM0338_BASIC, true)) {
                 $this->size++;
             } elseif (in_array($char, self::GSM0338_EXTENDED, true)) {
+                // In cases where a double counted char straddles two messages, add padding to push it to the next part
+                if (($this->size + 2) % self::MAXIMUM_CHARACTERS_7BIT_CONCATENATED === 1) {
```

Comment:
I haven't verified this, but I'm concerned that if the extended character is encountered at pos 153 (so also using 154) in a message whose total length that is shorter than 160, this will be adding unnecessary size.

E.g. number of characters is 159, one extended char at pos 153 with size counted as 3, pushes total size to 161 where actually it would be 160 and fit in a single message.

There should be a test for this.